program/src/process_management.rs

```diff
@@ -84,9 +84,11 @@ pub fn process_remove_validator(
     let mut lido = deserialize_lido(program_id, accounts.lido)?;
     lido.check_manager(accounts.manager)?;
 
-    let removed_validator = lido
-        .validators
-        .remove(accounts.validator_vote_account_to_remove.key)?;
+    let removed_validator = lido.validators.remove_on(
+        accounts.validator_vote_account_to_remove.key,
+        |x| x.can_be_removed(),
+        LidoError::ValidatorIsStillActive,
+    )?;
 
     if removed_validator.fee_credit != StLamports(0) {
```

Comment:
Maybe for defensive programming, we can assert that `validator.stake_accounts_balance == Lamports(0)` before we remove the validator. If it has no stake accounts, then its stake accounts balance *should* be 0, but if for whatever reason that is not the case, it is probably better to fail the transaction and keep the validator so we can investigate, than to allow it to succeed and possibly lose track of some SOL.